packages/astro/test/content-collections.test.js

```diff
@@ -222,6 +222,24 @@ describe('Content Collections', () => {
 				root: './fixtures/content-ssr-integration/',
 				output: 'server',
 				adapter: testAdapter(),
+				vite: {
+					plugins: [
+						// Verifies that `astro:content` does not have a hard dependency on Node builtins.
+						// This is to verify it will run on Cloudflare and Deno
+						{
+							name: 'verify-no-node-stuff',
+							generateBundle() {
+								const nodeModules = ['node:fs', 'node:url', 'node:worker_threads', 'node:path'];
+								nodeModules.forEach(name => {
+									const mod = this.getModuleInfo(name);
+									if(mod) {
+										throw new Error(`Node builtins snuck in: ${name}`)
+									}
+								});
```

Comment:
This is very smart! I wonder if this would be useful as a more general (not CC-specific) test as well?

Reply:
Totally. I remember multiple breaking bugs with the Markdown plugin a while back too